Sources/GRPCHTTP2Core/GRPCStreamStateMachine.swift

```diff
@@ -1469,6 +1521,43 @@ extension GRPCStreamStateMachine {
       return .awaitMoreMessages
     }
   }
+
+  private func getServerUnexpectedInboundCloseAction(
+    for reason: UnexpectedInboundCloseReason
+  ) -> OnUnexpectedInboundClose {
```

Comment:
Any reason this shouldn't be an `init` on `OnUnexpectedInboundClose`?

Reply:
Hmm I ended up doing this here because we're using this on the server side only (whereas the type itself is also used on the client side logic). But that's probably a not good enough reason, so I can move it.